File: oss-s3/src/main/java/dev/ocpd/oss/config/AwsS3Properties.java
package dev.ocpd.oss.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.lang.Nullable;

import java.net.URI;

@ConfigurationProperties("oss.s3")
public record AwsS3Properties(
    String accessKey,
    String secretKey,
    String region,
    String bucket,
    @Nullable
    URI endpoint) {

}

File: oss-s3/src/main/java/dev/ocpd/oss/config/AwsS3Properties.kt
package dev.ocpd.oss.config

import org.springframework.boot.context.properties.ConfigurationProperties
import java.net.URI

@ConfigurationProperties("oss.s3")
data class AwsS3Properties(
    val accessKey: String?,
    val secretKey: String?,
    val region: String,
    val bucket: String,
    val endpoint: URI?
)
